### DISCUSSION TITLE
`authStore.isValid` returns false immediately after login

### DISCUSSION THREAD

**Discussion by peterhijma** [pocketbase/pocketbase#2878] [Q&A]
Hi there, thanks for pocketbase.

I'm using OAuth (GitHub) in combination with Remix (SSR). I'm using a cookie to store the token.

Anyhow, I've got this problem where everything works as expected when the user was already logged in into GitHub. Immediately after logging in, `authStore.isValid` returns `true`. 
But **if** the user was not logged into GitHub yet, and the user therefore needs to complete the authentication flow first, the value of `authStore.isValid` is `false`. If I press F5 afterwards, the value is `true` again.

Any clue where I should be looking? 🤔 I can work around it probably, but I'd like to know what's going on.

**Comment by ganigeorgiev** [comment]
Are you sure that the cookie with the auth state is written to the user response?

You can try to export the cookie with `sameSite: 'Lax'` (see https://github.com/pocketbase/pocketbase/discussions/903#discussioncomment-4369458), but make sure to [read about the difference between `Strict` and `Lax`](https://web.dev/samesite-cookies-explained/#explicitly-state-cookie-usage-with-the-samesite-attribute)

**Reply by peterhijma** [reply]
It's already exported with `sameSite: 'Lax'`. It wouldn't explain the difference in behaviour between "already being logged in into Github" and "not yet being logged in, in GitHub" as well. (I think it's actually not about being logged in, but more about the authorizion step). I'll look into it a bit more later today. 

Thanks for the very quick response. 🙏

**Reply by ganigeorgiev** [reply]
Do you initialize the PocketBase SDK only server-side or maintain 2 JS SDK instances (browser and server)? If it is the latter, the SDK in the browser uses `localStorage` which could explain why after refreshing the page you are getting a loaded `authStore` state.

Could you provide a **minimal** reproducible code sample?

**Comment by peterhijma** [comment]
I'm only initialising it server side. But looking a bit into it more, it seems more a problem with (how I use...) Remix and cookies etc. then it is with pocketbase. Thanks for the quick response and I'll close this for now.

**Reply by peterhijma** [reply]
@ganigeorgiev I wasn't using `"lax"` 🤦 It works now.
